_posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md

```diff
@@ -0,0 +1,294 @@
+source projects can generally delay difficult decisions to make fast upfront
+progress to meet certain deadlines. This may be seen as an advantage at first,
+but as many developers can attest, this simply leads to technical debt and
+fragile products in most environments that struggle to prioritize a healthy
+codebase.
+
+Regardless, having to face these larger discussions upfront can induce fatigue,
+especially when managing external circumstances; personal affairs, a project at
+work - you know, the entity that pays these engineers - or countless other
+factors will rear their ugly heads and
+[progress will stagger](https://github.com/trinodb/trino/pull/17909#discussion_r1418149737)
+with ebbs and flows of attention. This can be really dangerous territory and
+commonly resolves in contributors and reviewers abandoning the PR when it stalls.
+
+This is why I believe open source, while not beholden to any timelines, needs a
+sort of project/product management role which is currently covered often by
+project leaders and DevEx engineers. This can also relieve tension between the
+needs of open source and big businesses in the community with real deadlines, at
+least keeping the communication consistent while ensuring bugs and design flaws
+aren’t introduced to the code base.
+
+## What’s in it for Bloomberg and ForePaaS?
+
+If you’ve never worked in open source or for a company that contributes to open
+source, you may be thinking how the heck do these engineers convince their
+leadership to let them dump so much time into these contributions? The simple
+answer is, it’s good for business.
+
+If we peep into why Bloomberg uses Trino, they aggregate data from an unusually
+large number of data sources across their customers who use their services. Part
+of this requires them to merge the customer’s dataset with existing aggregate
+data in Bloomberg’s product. Since Trino can connect to most customer databases
+out-of-the-box, this requires Bloomberg to manage a small array of custom
+connectors that provide their services to customers as multiple catalogs in a
+single convention SQL endpoint. Having engineers maintain a few small connectors
+rather than an entire distributed query engine themselves saves a lot of time
+and maintenance.
+
+Despite how many problems Trino already solves for them, Bloomberg ideally still
+wants to maintain as few things as possible to relieve their engineer’s time.
+Luckily, open source projects are generally more than happy to accept features
+that the community all benefit from. This doesn’t mean we shouldn’t appreciate
+when companies contribute. This generosity and forward-thinking approach enabled
+Erik and Teng to combine their battle-tested connectors, crafting a high value
+creation for the community.
+
+Yet another issue that has come up much more recently is the
+[XZ exploit](https://www.darkreading.com/application-security/xz-utils-scare-exposes-hard-truths-in-software-security)
```

Comment:
I'm not sure the tangent of security exploits really belongs in this blog post - or at least it isn't an issue that the Trino community would be super concerned with, I think.